src/cpu.rs

```diff
@@ -476,26 +555,234 @@ impl Cpu {
+                self.sp = self.sp.wrapping_sub(2);
+            }
+            Operand::D => {
+                let res = self.registers.get_de();
+                self.memory[self.sp as usize - 1] = res as u8;
+                self.memory[self.sp as usize - 2] = (res >> 8) as u8;
+                self.sp = self.sp.wrapping_sub(2);
+            }
+            Operand::H => {
+                let res = self.registers.get_hl();
+                self.memory[self.sp as usize - 1] = res as u8;
+                self.memory[self.sp as usize - 2] = (res >> 8) as u8;
+                self.sp = self.sp.wrapping_sub(2);
+            }
+            Operand::PSW => {
+                self.memory[self.sp as usize - 1] = self.registers.a;
+                self.memory[self.sp as usize - 2] = self.condition_codes.flags_to_psw();
+                self.sp = self.sp.wrapping_sub(2);
+            }
+            _ => {
+                //TODO: write error message later
+                unimplemented!();
+            }
+        };
+    }
+
+    // The contents of the specified register pair are restored from two
+    // bytes of memory indicated by the stack pointer SP.
+    // Condition bits affected: None
+    fn pop(&mut self, reg: Operand) {
+        match reg {
+            Operand::B => {
+                self.registers.b = self.memory[self.sp as usize];
+                self.registers.c = self.memory[self.sp as usize + 1];
+                self.sp = self.sp.wrapping_add(2);
+            }
+            Operand::D => {
+                self.registers.d = self.memory[self.sp as usize];
+                self.registers.e = self.memory[self.sp as usize + 1];
+                self.sp = self.sp.wrapping_add(2);
+            }
+            Operand::H => {
+                self.registers.h = self.memory[self.sp as usize];
+                self.registers.l = self.memory[self.sp as usize + 1];
+                self.sp = self.sp.wrapping_add(2);
+            }
+            Operand::PSW => {
+                self.registers.a = self.memory[self.sp as usize];
+                let res = self.memory[self.sp as usize + 1];
+                self.condition_codes.psw_to_flags(res);
+                self.sp = self.sp.wrapping_add(2);
+            }
+            _ => {
+                //TODO: write error message later
+                unimplemented!();
+            }
+        };
     }
 
-    fn pop(&self) -> u16 {
-        // TODO
-        0
+    // Double Add. The 16-bit number in the specified register pair is added to the
+    // 16-bit number held in the H and L registers using two's complement arithmetic.
+    // The result replaces the contents of the H and L registers.
+    // Condition bits affected: Carry
+    fn dad(&mut self, reg: Operand) {
+        match reg {
+            Operand::B => {
+                let res = self.registers.get_bc();
+                self.condition_codes
+                    .set_carry((res + self.registers.get_hl()) > 0xFF);
```

Comment:
I updated these comparisons to 0xFF as the comparison would not be evaluated since it would previously always evaluate to false.